src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -279,12 +281,19 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
           style={{ backdropFilter: 'blur(24px)' }}
         >
           <Row marginTop="12" gap="12">
-            <FilterButton
-              isMobile={isMobile}
-              isFiltersExpanded={isFiltersExpanded}
-              onClick={() => setFiltersExpanded(!isFiltersExpanded)}
-              collectionCount={collectionNfts?.[0]?.totalCount ?? 0}
-            />
+            <TraceEvent
+              events={[Event.onClick]}
+              element={ElementName.NFT_FILTER_BUTTON}
+              name={EventName.NFT_FILTER_OPENED}
```

Comment:
Can we change this to NFT_FILTER_TOGGLED as this fires whenever the filter is opened/closed. Alternatively only log the event if the Filter was closed prior to click

Reply:
at the time I wasn't aware of `shouldLogImpression` - using that now to conditionally fire events